File: reference/springboot/domain-usecases/src/main/java/org/example/domain/usecases/types/Dog.java
package org.example.domain.usecases.types;

import lombok.Builder;
import lombok.Getter;

import java.util.UUID;

@Getter
public class Dog extends Animal {
    private String a;

    @Builder
    public Dog(UUID id, String sound, String a) {
        super(id, sound);
        this.a = a;
    }

    @Override
    public <T> T visit(AnimalVisitor<T> visitor) {
        return visitor.visitDog(this);
    }
}

File: reference/dotnet/Domain/Company.Product.Domain.UseCases/Types/Dog.cs
using System;

namespace Company.Product.Domain.UseCases.Types
{
    public class Dog : Animal
    {
        public string A { get;}

        public Dog(Guid animalId, string sound, string a) : base(animalId, sound)
        {
            A = a;
        }

        public override T Visit<T>(AnimalVisitor<T> visitor)
        {
            if (visitor is null)
            {
                throw new ArgumentNullException(nameof(visitor));
            }

            return visitor.VisitDog(this);
        }
    }
}